aptos-move/framework/aptos-framework/sources/delegation_pool.move

```diff
@@ -722,6 +722,22 @@ module aptos_framework::delegation_pool {
         )
     }
 
+    #[view]
+    /// Return the current state of a voting delegation of a delegator in a delegation pool.
+    public fun calculate_and_update_voting_delegation(
+        pool_address: address,
+        delegator_address: address
+    ): (address, address, u64) acquires DelegationPool, GovernanceRecords {
+        assert_partial_governance_voting_enabled(pool_address);
+        let vote_delegation = update_and_borrow_mut_delegator_vote_delegation(
+            borrow_global<DelegationPool>(pool_address),
+            borrow_global_mut<GovernanceRecords>(pool_address),
+            delegator_address
+        );
+
+        (vote_delegation.voter, vote_delegation.pending_voter, vote_delegation.last_locked_until_secs)
```

Comment:
You can also return a copy of the entire vote_delegation struct here

Reply:
A copy was returned initially, but object's fields could not be accessed within modules other than `delegation_pool` where `VoteDelegation` is defined. This version accommodates callers both on-chain and off-chain.